build.py

```diff
@@ -229,11 +229,11 @@ def _validate_build_dir(args: argparse.Namespace):
             # also tweak build directory name for mac builds
             target_sys = "macOS"
 
-        args.build_dir = Path("build") / target_sys
+        args.build_dir = Path("build") / target_sys / args.config
```

Comment:
This makes it easier for `build.py` reuse the same output dir as `cmake --preset`.